facepy/graph_api.py

```diff
@@ -152,33 +152,41 @@ def batch(self, requests):
         """
         Make a batch request.
 
-        :param requests: A list of dictionaries with keys 'method', 'relative_url' and optionally 'body'.
+        :param requests: A list or list comprehension of dictionaries with keys 'method',
+                         'relative_url' and optionally 'body'.
 
-        Yields a list of responses and/or exceptions.
+        Yields a generator of responses and/or exceptions.
         """
 
-        for request in requests:
-            if 'body' in request:
-                request['body'] = urlencode(request['body'])
-
         def _grouper(complete_list, n=1):
             """
-            Batches a list into constant size chunks.
+            Batches a list or generator into constant size chunks.
```

Comment:
generator is a specific type (the type of objects returned by functions that use yield), the generic concept here is iterable.